src/Yesod/Auth/OAuth2/ErrorResponse.hs

```diff
@@ -32,6 +34,25 @@ data ErrorResponse = ErrorResponse
     }
     deriving Show
 
+-- | Textual value suitable for display to a User
+erUserMessage :: ErrorResponse -> Text
+erUserMessage err = case erName err of
+    InvalidRequest -> "Invalid request"
+    UnauthorizedClient -> "Unauthorized client"
+    AccessDenied -> "Access denied"
+    UnsupportedResponseType -> "Unsupported response type"
+    InvalidScope -> "Invalid scope"
+    ServerError -> "Server error"
+    TemporarilyUnavailable -> "Temporarily unavailable"
+    Unknown _ -> "Unknown error"
+
+unknownError :: Text -> ErrorResponse
+unknownError x = ErrorResponse
+    { erName = Unknown x
```

Comment:
As an alternative, I could dump this `x` into `Description` instead, and leave the `Name` as a value-less `Unknown`. Then `erUserMessage` may be unneccessary as a simple `show`-like on `ErrorName` couldn't contain any details. :man_shrugging: